mindsdb/integrations/mysql/mysql.py

```diff
@@ -0,0 +1,149 @@
+import mysql.connector
+
+from mindsdb_native.libs.constants.mindsdb import DATA_SUBTYPES
+
+
+class MySQL():
+    def __init__(self, config, name):
+        self.config = config
+        self.name = name
+
+    def _to_mysql_table(self, stats, predicted_cols):
+        subtype_map = {
+            DATA_SUBTYPES.INT: 'int',
+            DATA_SUBTYPES.FLOAT: 'double',
+            DATA_SUBTYPES.BINARY: 'bool',
+            DATA_SUBTYPES.DATE: 'Date',
+            DATA_SUBTYPES.TIMESTAMP: 'Datetime',
+            DATA_SUBTYPES.SINGLE: 'VARCHAR(500)',
+            DATA_SUBTYPES.MULTIPLE: 'VARCHAR(500)',
+            DATA_SUBTYPES.IMAGE: 'VARCHAR(500)',
+            DATA_SUBTYPES.VIDEO: 'VARCHAR(500)',
+            DATA_SUBTYPES.AUDIO: 'VARCHAR(500)',
+            DATA_SUBTYPES.SHORT: 'VARCHAR(500)',
+            DATA_SUBTYPES.RICH: 'VARCHAR(500)',
+            DATA_SUBTYPES.ARRAY: 'VARCHAR(500)'
+        }
+
+        column_declaration = []
+        for name, column in stats.items():
+            try:
+                col_subtype = stats[name]['typing']['data_subtype']
+                new_type = subtype_map[col_subtype]
+                column_declaration.append(f' `{name}` {new_type} ')
+                if name in predicted_cols:
+                    column_declaration.append(f' `{name}_original` {new_type} ')
+            except Exception:
+                print(f'Error: cant convert type {col_subtype} of column {name} to mysql type')
+
+        return column_declaration
+
+    def _escape_table_name(self, name):
+        return '`' + name.replace('`', '``') + '`'
+
+    def _query(self, query):
+        con = mysql.connector.connect(host=self.config['integrations'][self.name]['host'], port=self.config['integrations'][self.name]['port'], user=self.config['integrations'][self.name]['user'], password=self.config['integrations'][self.name]['password'])
+
+        cur = con.cursor(dictionary=True, buffered=True)
+        cur.execute(query)
+        res = True
+        try:
+            res = cur.fetchall()
+        except Exception:
+            pass
+        con.commit()
+        con.close()
+
+        return res
+
+    def _get_connect_string(self, table):
+        user = f"{self.config['api']['mysql']['user']}_{self.name}"
+        password = self.config['api']['mysql']['password']
+        host = self.config['api']['mysql']['host']
+        port = self.config['api']['mysql']['port']
+
+        if password is None or password == '':
+            connect = f'mysql://{user}@{host}:{port}/mindsdb/{table}'
+        else:
+            connect = f'mysql://{user}:{password}@{host}:{port}/mindsdb/{table}'
+
+        return connect
+
+    def setup(self):
+        self._query('DROP DATABASE IF EXISTS mindsdb')
+
+        self._query('CREATE DATABASE IF NOT EXISTS mindsdb')
+
+        connect = self._get_connect_string('predictors')
+
+        q = f"""
+                CREATE TABLE IF NOT EXISTS mindsdb.predictors
+                (name VARCHAR(500),
```

Comment:
I'm seeing this `"VARCHAR(500)"` in multiple files, it might make sense to declare a variable in constants like `LONG_VARCHAR = "VARCHAR(500)"` and use it everywhere else, it will be easier to change this magic value `500` in future

Reply:
it used in 'mariadb' and 'mysql'. I will change this when whill change this files in next time